webrtc/README.md

```diff
@@ -0,0 +1,268 @@
+# WebRTC
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Working Draft | Active |                 |
+
+Authors: [@mxinden]
+
+Interest Group: [@marten-seemann]
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+**Table of Contents**
+
+- [WebRTC](#webrtc)
+    - [Motivation](#motivation)
+    - [Requirements](#requirements)
+    - [Addressing](#addressing)
+    - [Connection Establishment](#connection-establishment)
+        - [Browser to public Server](#browser-to-public-server)
+            - [Open Questions](#open-questions)
+        - [Browser to Browser](#browser-to-browser)
+            - [Open Questions](#open-questions-1)
+    - [Connection Security](#connection-security)
+        - [Open Questions](#open-questions-2)
+    - [Multiplexing](#multiplexing)
+    - [General Open Questions](#general-open-questions)
+    - [Previous, ongoing and related work](#previous-ongoing-and-related-work)
+- [FAQ](#faq)
+
+<!-- markdown-toc end -->
+
+
+## Motivation
+
+1. **No need for valid TLS certificates.** Enable browsers to connect to public
+   server nodes without those server nodes providing a TLS certificate within
+   the browsers trustchain. Note that we can not do this today with our
+   Websocket transport.
+
+2. **Hole punching in the browser**: Enable two browsers or a browser and a
+   non-public server node to connect.
+
+## Requirements
+
+- Loading a remote nodes certificate into ones browser trust-store is not an
+  option, i.e. doesn't scale.
+
+- No dependency on central STUN and/or TURN servers.
+
+## Addressing
+
+WebRTC multiaddresses are composed of an IP and UDP address component, followed
+by `/webrtc` and a multihash of the certificate that the node uses.
+
+Examples:
+- `/ip4/1.2.3.4/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>`
```

Comment:
Is this the UDP address of the STUN server? We should probably mention here that the actual WebRTC data channel won't use UDP and this port.

Reply:
Adding to the response by @ckousik above:

> We should probably mention here that the actual WebRTC data channel won't use UDP and this port.

The actual WebRTC data channel will be using UDP and this port. WebRTC can run on top of TCP (see below) though prefers UDP. WebRTC data channels use SCTP running on top of DTLS running on top of UDP.

Reply:
Resolving here. @marten-seemann please comment in case you want this further documented.